src/App.tsx

```diff
@@ -8,37 +8,111 @@ import { PostsList } from './components/PostsList';
+  const [isLoading, setIsLoading] = useState(false);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [error, setError] = useState(Errors.NONE);
+  const [isLoadingComments, setIsLoadingComments] = useState(false);
+
+  const loadUsers = async () => {
+    try {
+      const usersFromServer = await getUsers();
+
+      setUsers(usersFromServer);
+    } catch {
+      setError(Errors.USERS);
+    }
+  };
+
+  const loadSelectedUserPosts = async () => {
+    if (selectedUser) {
+      try {
+        setIsLoading(true);
+        const posts = await getPostsOfUser(selectedUser.id);
+
+        setUserPosts(posts);
+      } catch {
+        setError(Errors.POSTS);
+      } finally {
+        setIsLoading(false);
+      }
+    }
+  };
+
+  useEffect(() => {
+    loadUsers();
+  }, []);
+
+  useEffect(() => {
+    loadSelectedUserPosts();
+  }, [selectedUser]);
+
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector
+                  users={users}
+                  selectedUser={selectedUser}
+                  setSelectedUser={setSelectedUser}
+                  error={error}
+                  setSelectedPost={setSelectedPost}
+                />
               </div>
 
               <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">
-                  No user selected
-                </p>
-
-                <Loader />
+                {isLoading
+                  ? <Loader />
+                  : (
+                    <>
+                      {(error === Errors.POSTS || error === Errors.USERS) && (
+                        <div
+                          className="notification is-danger"
+                          data-cy="PostsLoadingError"
+                        >
+                          {error}
+                        </div>
+                      )}
 
-                <div
-                  className="notification is-danger"
-                  data-cy="PostsLoadingError"
-                >
-                  Something went wrong!
-                </div>
+                      {!selectedUser && error === Errors.NONE && (
+                        <p data-cy="NoSelectedUser">
+                          No user selected
+                        </p>
+                      )}
 
-                <div className="notification is-warning" data-cy="NoPostsYet">
-                  No posts yet
-                </div>
+                      {selectedUser
+                      && !userPosts.length
+                      && error !== Errors.POSTS
+                      && (
+                        <div
+                          className="notification is-warning"
```

Comment:
looks like you need to fix indentions here
and instead of multiple rows of conditions - better to pass them into variable and use variable